packages/buidler-core/src/builtin-tasks/compile.ts

```diff
@@ -250,6 +244,24 @@ async function removeObsoleteArtifacts(
   }
 }
 
+async function removeObsoleteBuildInfos(artifactsPath: string) {
```

Comment:
I will add a comment to this function. But this is what it does: it goes through all the `dbg` files and builds a set of all the referenced buildInfo files. Then it removes the rest.